### PR TITLE
[Icons] Upgrading to FontAwesome v4.5.0

### PR DESCRIPTION
**Pull request by BreadMaker**
tbh idk if i'm doing it right.

But here is it.

### CODE REVIEW

File: src/themes/default/elements/icon.overrides [hunk 1]
```diff
@@ -137,6 +148,16 @@ i.icon.add.to.cart:before { content: "\f217"; }
 i.icon.in.cart:before { content: "\f218"; }
 i.icon.add.user:before { content: "\f234"; }
 i.icon.remove.user:before { content: "\f235"; }
+i.icon.object.group:before { content: "\f247"; }
+i.icon.object.ungroup:before { content: "\f248"; }
+i.icon.clone:before { content: "\f24d"; }
+i.icon.hand.rock:before { content: "\f255"; }
```
Comment (BreadMaker) [inline src/themes/default/elements/icon.overrides]: This new subset makes ref to "Rock-Paper-Scissors-Lizard-Spock". It's OK to put them under the User Actions category?

File: src/themes/default/elements/icon.overrides [hunk 2]
```diff
@@ -590,6 +634,29 @@ i.icon.skyatlas:before { content: "\f216"; }
 i.icon.whatsapp:before { content: "\f232"; }
 i.icon.viacoin:before { content: "\f237"; }
 i.icon.medium:before { content: "\f23a"; }
+i.icon.y.combinator:before { content: "\f23b"; }
+i.icon.optinmonster:before { content: "\f23c"; }
+i.icon.opencart:before { content: "\f23d"; }
+i.icon.expeditedssl:before { content: "\f23e"; }
+i.icon.gg:before { content: "\f260"; }
```
Comment (BreadMaker) [inline src/themes/default/elements/icon.overrides]: I don't know what this "GG" stands for.

File: src/themes/default/elements/icon.overrides [hunk 3]
```diff
@@ -590,6 +634,29 @@ i.icon.skyatlas:before { content: "\f216"; }
 i.icon.whatsapp:before { content: "\f232"; }
 i.icon.viacoin:before { content: "\f237"; }
 i.icon.medium:before { content: "\f23a"; }
+i.icon.y.combinator:before { content: "\f23b"; }
```
Comment (BreadMaker) [inline src/themes/default/elements/icon.overrides]: What would be the best: "y combinator icon" or "ycombinator icon"?

File: src/themes/default/elements/icon.overrides [hunk 4]
```diff
@@ -137,6 +148,16 @@ i.icon.add.to.cart:before { content: "\f217"; }
...
 i.icon.add.user:before { content: "\f234"; }
 i.icon.remove.user:before { content: "\f235"; }
+i.icon.object.group:before { content: "\f247"; }
+i.icon.object.ungroup:before { content: "\f248"; }
+i.icon.clone:before { content: "\f24d"; }
+i.icon.hand.rock:before { content: "\f255"; }
+i.icon.hand.paper:before { content: "\f256"; }
+i.icon.hand.scissors:before { content: "\f257"; }
+i.icon.hand.lizard:before { content: "\f258"; }
+i.icon.hand.spock:before { content: "\f259"; }
```
Comment (BreadMaker) [inline src/themes/default/elements/icon.overrides]: This new subset makes ref to "Rock-Paper-Scissors-Lizard-Spock". It's OK to put them under the User Actions category?

**Comment by jlukic** [comment]
Hey you're doing it exactly right. Group by category and sort by UTF code :+1: 

Generally I add aliases for any icons which names are often confused. If you do that just add another reference to the icon under "Aliases"

After we can look at doing a PR for Docs.

**Comment by jlukic** [comment]
#2902 #2857

**Comment by BreadMaker** [comment]
All righty then. 

I'll double check if there's missing icons, and for more aliases as suggested.

File: src/themes/default/elements/icon.overrides [hunk 5]
```diff
@@ -753,3 +820,26 @@ i.icon.mars.stroke.horizontal:before { content: "\f22b"; }
 i.icon.facebook.official { content: "\f230"; }
 i.icon.pinterest.official { content: "\f231"; }
 i.icon.bed:before { content: "\f236"; }
+i.icon.yc:before { content: "\f23b"; }
```
Comment (BreadMaker) [inline src/themes/default/elements/icon.overrides]: The alias of ycombinator it's here.

**Comment by jholl** [comment]
if it's any help -- I marked them [here](https://github.com/jholl/Semantic-UI/blob/font_awesome_4.4/src/themes/default/elements/icon.overrides) in UTF order as things unique to semantic (aliases) and things that cause a breaking conflict (because Semantic uses a word that 4.4 now also uses, i.e. `map`)

**Comment by BreadMaker** [comment]
Good one! I'll take a look.

**Comment by BreadMaker** [comment]
@jholl yes, it was very useful. There's another one to add: Now there's a new icon `creative commons` (f25e), whose alias is `cc`, but it's taken by `closed captioning` (f20a).

**Comment by BreadMaker** [comment]
I'll update new changes soon. I'm triple checking everything. I'm afraid there will be some breaking changes.

**Comment by jlukic** [comment]
We cant add any breaking changes..

**Comment by BreadMaker** [comment]
I'm doing my best to prevent any breaking change. That's why I'm taking so long.

**Comment by jlukic** [comment]
Keep in mind css class names cant start with numbers as well.

**Comment by BreadMaker** [comment]
Ok, here we go.

File: src/themes/default/elements/icon.overrides [hunk 6]
```diff
@@ -537,7 +582,7 @@ i.icon.weibo:before { content: "\f18a"; }
 i.icon.renren:before { content: "\f18b"; }
 i.icon.pagelines:before { content: "\f18c"; }
 i.icon.stack.exchange:before { content: "\f18d"; }
-i.icon.vimeo:before { content: "\f194"; }
+i.icon.vimeo.square:before { content: "\f194"; }
```
Comment (BreadMaker) [inline src/themes/default/elements/icon.overrides]: With the addition of the new `vimeo` (the actual v) icon, makes more sense that the former icon now refers to `square vimeo`. That is my proposal.

File: src/themes/default/elements/icon.overrides [hunk 7]
```diff
@@ -428,6 +466,10 @@ i.icon.street.view:before { content: "\f21d"; }
 i.icon.hotel:before { content: "\f236"; }
 i.icon.train:before { content: "\f238"; }
 i.icon.subway:before { content: "\f239"; }
+i.icon.map.pin:before { content: "\f276"; }
+i.icon.map.signs:before { content: "\f277"; }
+i.icon.map.outline:before { content: "\f278"; }
+i.icon.map:before { content: "\f279"; }
```
Comment (BreadMaker) [inline src/themes/default/elements/icon.overrides]: Now there's an icon for `map`, so the previous alias to `compass` should be changed.

File: src/themes/default/elements/icon.overrides [hunk 8]
```diff
@@ -587,9 +632,34 @@ i.icon.sellsy:before { content: "\f213"; }
 i.icon.shirtsinbulk:before { content: "\f214"; }
 i.icon.simplybuilt:before { content: "\f215"; }
 i.icon.skyatlas:before { content: "\f216"; }
+i.icon.facebook:before { content: "\f230"; }
+i.icon.pinterest:before { content: "\f231"; }
```
Comment (BreadMaker) [inline src/themes/default/elements/icon.overrides]: btw, these icons were not present due a typo.

File: src/themes/default/elements/icon.overrides [hunk 9]
```diff
@@ -747,9 +822,42 @@ i.icon.venus.mars:before { content: "\f228"; }
 i.icon.mars.stroke:before { content: "\f229"; }
 i.icon.mars.alternate:before { content: "\f229"; }
 i.icon.mars.vertical:before { content: "\f22a"; }
-i.icon.mars.horizontal:before { content: "\f22b"; }
 i.icon.mars.stroke.vertical:before { content: "\f22a"; }
+i.icon.mars.horizontal:before { content: "\f22b"; }
 i.icon.mars.stroke.horizontal:before { content: "\f22b"; }
-i.icon.facebook.official { content: "\f230"; }
-i.icon.pinterest.official { content: "\f231"; }
```
Comment (BreadMaker) [inline src/themes/default/elements/icon.overrides]: right here

File: src/themes/default/elements/icon.overrides (hunk 7 above)
Comment (BreadMaker) [inline src/themes/default/elements/icon.overrides]: In fact, it is changed in this PR.

`map` (before `f14e`, was alias of `compass`) it's now `f279` (the actual icon)

File: src/themes/default/elements/icon.overrides [hunk 10]
```diff
@@ -173,6 +194,7 @@ i.icon.other.gender:before { content: "\f229"; }
 i.icon.other.gender.vertical:before { content: "\f22a"; }
 i.icon.other.gender.horizontal:before { content: "\f22b"; }
 i.icon.neuter:before { content: "\f22c"; }
+i.icon.genderless:before { content: "\f22d"; }
```
Comment (BreadMaker) [inline src/themes/default/elements/icon.overrides]: Now `genderless` uses the proper icon, not an alias for `thin circle`

File: src/themes/default/elements/icon.overrides [hunk 11]
```diff
@@ -705,40 +771,49 @@ i.icon.target:before { content: "\f140"; }
...
 i.icon.cc:before { content: "\f20a"; }
+i.icon.sheqel:before { content: "\f20b"; }
+i.icon.ils:before { content: "\f20b"; }
 i.icon.plus.cart:before { content: "\f217"; }
 i.icon.arrow.down.cart:before { content: "\f218"; }
 i.icon.detective:before { content: "\f21b"; }
 i.icon.venus:before { content: "\f221"; }
 i.icon.mars:before { content: "\f222"; }
 i.icon.mercury:before { content: "\f223"; }
+i.icon.intersex:before { content: "\f224"; }
```
Comment (BreadMaker) [inline src/themes/default/elements/icon.overrides]: Now `intersex` uses the proper icon, not an alias for `outline circle`

**Comment by BreadMaker** [comment]
I think that's all. btw, thanks @jholl

File: src/themes/default/elements/icon.overrides [hunk 12]
```diff
@@ -412,6 +450,7 @@ i.icon.first.aid:before { content: "\f0fa"; }
 i.icon.military:before { content: "\f0fb"; }
 i.icon.h:before { content: "\f0fd"; }
 i.icon.location.arrow:before { content: "\f124"; }
+i.icon.compass:before { content: "\f14e"; }
```
Comment (BreadMaker) [inline src/themes/default/elements/icon.overrides]: With the addition of `map`, `compass` now can be in the Map category.

File: src/themes/default/elements/icon.overrides [hunk 13]
```diff
@@ -480,18 +524,21 @@ i.icon.rupee:before { content: "\f156"; }
 i.icon.yen:before { content: "\f157"; }
 i.icon.ruble:before { content: "\f158"; }
 i.icon.won:before { content: "\f159"; }
+i.icon.bitcoin:before { content: "\f15a"; }
```
Comment (BreadMaker) [inline src/themes/default/elements/icon.overrides]: Should `bitcoin` be in the Currency category? Please, share you thoughts.

**Comment by BreadMaker** [comment]
I'll keep combing the file, looking for inconsistencies.

File: src/themes/default/elements/icon.overrides (hunk 13 above)
Comment (jlukic) [inline src/themes/default/elements/icon.overrides]: Yea

File: src/themes/default/elements/icon.overrides (hunk 11 above)
Comment (jlukic) [inline src/themes/default/elements/icon.overrides]: :+1:

File: src/themes/default/elements/icon.overrides (hunk 10 above)
Comment (jlukic) [inline src/themes/default/elements/icon.overrides]: :

File: src/themes/default/elements/icon.overrides (hunk 7 above)
Comment (jlukic) [inline src/themes/default/elements/icon.overrides]: That seems good.

**Comment by jlukic** [comment]
Everything set here?

**Comment by BreadMaker** [comment]
We are good to go, @jlukic

**Comment by zuck** [comment]
Font Awesome 4.5 is out now...

**Comment by BreadMaker** [comment]
Adding them is straightforward. Just give me a moment.

File: src/themes/default/elements/icon.overrides [hunk 14]
```diff
@@ -587,9 +646,42 @@ i.icon.sellsy:before { content: "\f213"; }
...
+i.icon.opera:before { content: "\f26a"; }
+i.icon.internet.explorer:before { content: "\f26b"; }
+i.icon.contao:before { content: "\f26d"; }
+i.icon.five.hundred.pixels:before { content: "\f26e"; }
+i.icon.amazon:before { content: "\f270"; }
+i.icon.houzz:before { content: "\f27c"; }
+i.icon.vimeo:before { content: "\f27d"; }
+i.icon.black.tie:before { content: "\f27e"; }
+i.icon.fonticons:before { content: "\f280"; }
+i.icon.reddit.alien:before { content: "\f281"; }
```
Comment (BreadMaker) [inline src/themes/default/elements/icon.overrides]: `reddit alien`, very straightforward imho

File: src/themes/default/elements/icon.overrides [hunk 15]
```diff
@@ -747,9 +842,45 @@ i.icon.venus.mars:before { content: "\f228"; }
...
+i.icon.500px:before { content: "\f26e"; }
+i.icon.calendar.plus:before { content: "\f271"; }
+i.icon.calendar.minus:before { content: "\f272"; }
+i.icon.calendar.times:before { content: "\f273"; }
+i.icon.calendar.check:before { content: "\f274"; }
+i.icon.factory:before { content: "\f275"; }
+i.icon.muted.user:before { content: "\f27a"; }
+i.icon.muted.user.outline:before { content: "\f27b"; }
+i.icon.edge:before { content: "\f282"; }
+i.icon.ms.edge:before { content: "\f282"; }
```
Comment (BreadMaker) [inline src/themes/default/elements/icon.overrides]: `microsoft edge` (under Brands category) it's the "official" name, being aliases `edge` and `ms edge`.

File: src/themes/default/elements/icon.overrides [hunk 16]
```diff
@@ -480,18 +536,22 @@ i.icon.rupee:before { content: "\f156"; }
...
 i.icon.google.wallet:before { content: "\f1ee"; }
 i.icon.visa:before { content: "\f1f0"; }
 i.icon.mastercard:before { content: "\f1f1"; }
 i.icon.discover:before { content: "\f1f2"; }
 i.icon.american.express:before { content: "\f1f3"; }
+i.icon.paypal.card:before { content: "\f1f4"; }
 i.icon.stripe:before { content: "\f1f5"; }
+i.icon.japan.credit.bureau:before { content: "\f24b"; }
+i.icon.diners.club:before { content: "\f24c"; }
+i.icon.credit.card.alternative:before { content: "\f283"; }
```
Comment (BreadMaker) [inline src/themes/default/elements/icon.overrides]: The new `alternative credit card` icon is under the Payment Options category.

File: src/themes/default/elements/icon.overrides [hunk 17]
```diff
@@ -353,16 +396,19 @@ i.icon.file.video.outline:before { content: "\f1c8"; }
...
 i.icon.html5:before { content: "\f13b"; }
 i.icon.css3:before { content: "\f13c"; }
-i.icon.rss:before { content: "\f09e"; }
 i.icon.rss.square:before { content: "\f143"; }
 i.icon.openid:before { content: "\f19b"; }
 i.icon.database:before { content: "\f1c0"; }
 i.icon.server:before { content: "\f233"; }
+i.icon.usb:before { content: "\f287"; }
+i.icon.bluetooth:before { content: "\f293"; }
+i.icon.bluetooth.alternative:before { content: "\f294"; }
```
Comment (BreadMaker) [inline src/themes/default/elements/icon.overrides]: Three new Technology icons: `usb`, `bluetooth` and `bluetooth alternative` (that last one is the "B" without its background ellipse)

File: src/themes/default/elements/icon.overrides [hunk 18]
```diff
@@ -82,8 +84,24 @@ i.icon.at:before { content: "\f1fa"; }
...
+i.icon.creative.commons:before { content: "\f25e"; }
+i.icon.add.to.calendar:before { content: "\f271"; }
+i.icon.remove.from.calendar:before { content: "\f272"; }
+i.icon.delete.calendar:before { content: "\f273"; }
+i.icon.checked.calendar:before { content: "\f274"; }
+i.icon.industry:before { content: "\f275"; }
+i.icon.shopping.bag:before { content: "\f290"; }
+i.icon.shopping.basket:before { content: "\f291"; }
+i.icon.hashtag:before { content: "\f292"; }
+i.icon.percent:before { content: "\f295"; }
```
Comment (BreadMaker) [inline src/themes/default/elements/icon.overrides]: New icons for the Web Content category: `shopping bag`, `shopping basket`, `hashtag` (the # char) and `percent` (yes, the % char)

File: src/themes/default/elements/icon.overrides [hunk 19]
```diff
@@ -400,6 +446,10 @@ i.icon.unmute:before { content: "\f130"; }
 i.icon.mute:before { content: "\f131"; }
 i.icon.video.play:before { content: "\f144"; }
 i.icon.closed.captioning:before { content: "\f20a"; }
+i.icon.pause.circle:before { content: "\f28b"; }
+i.icon.pause.circle.outline:before { content: "\f28c"; }
+i.icon.stop.circle:before { content: "\f28d"; }
+i.icon.stop.circle.outline:before { content: "\f28e"; }
```
Comment (BreadMaker) [inline src/themes/default/elements/icon.overrides]: New icons for the Audio category: `pause circle`, `pause circle outline`, `stop circle`, `stop circle outline`

File: src/themes/default/elements/icon.overrides [hunk 20]
```diff
@@ -587,9 +646,42 @@ i.icon.sellsy:before { content: "\f213"; }
...
+i.icon.black.tie:before { content: "\f27e"; }
+i.icon.fonticons:before { content: "\f280"; }
+i.icon.reddit.alien:before { content: "\f281"; }
+i.icon.microsoft.edge:before { content: "\f282"; }
+i.icon.codiepie:before { content: "\f284"; }
+i.icon.modx:before { content: "\f285"; }
+i.icon.fort.awesome:before { content: "\f286"; }
+i.icon.product.hunt:before { content: "\f288"; }
+i.icon.mixcloud:before { content: "\f289"; }
+i.icon.scribd:before { content: "\f28a"; }
```
Comment (BreadMaker) [inline src/themes/default/elements/icon.overrides]: Finally, a lot of new icons for the Networks and Websites category: `microsoft edge` (the browser [?]), `codiepie` (wordpress themes creators), `modx` (web CMS), `fort awesome` (the FontAwesome creators), `product hunt` (the product discovery site), `mixcloud` (the music streaming service), and `scribd` (the digital library)

**Comment by BreadMaker** [comment]
Ok, @jlukic, we are good to go. Again.

**Comment by jlukic** [comment]
@BreadMaker Im sorry for taking way too long with this, I've reviewed and like all the naming choices, great work.

Will be updating docs.

**Comment by jlukic** [comment]
Added to docs in 6f0fb514af1cdf808399ab03a930b54fc01eba4f

**Comment by BreadMaker** [comment]
OMG, yes! I'm so glad to contribute to this great project! Keep up with the awesome work, @jlukic!

**Comment by BreadMaker** [comment]
Also, props to @jholl for his list of aliases conflicts, that helped me a lot!

**Pull request merged**